### PR TITLE
interfaces: add some tests for the mount backend ns update case

### PR DESCRIPTION
**Pull request by pedronis**
This is also helpful as baseline to add unit tests in #12541

### CODE REVIEW

File: interfaces/mount/backend_test.go [hunk 1]
```diff
@@ -246,3 +246,107 @@ func (s *backendSuite) TestSandboxFeatures(c *C) {
...
+
+	update = true
+	// ensure .mnt file
+	mntFile := filepath.Join(dirs.SnapRunNsDir, "snap-name.mnt")
+	err := ioutil.WriteFile(mntFile, []byte(""), 0644)
+	c.Assert(err, IsNil)
+
+	// confinement options are irrelevant to this security backend
+	_, err = s.UpdateSnapMaybeErr(c, snapInfo, interfaces.ConfinementOptions{}, mockSnapYaml, 1)
+	c.Check(err, ErrorMatches, `cannot update mount namespace of snap "snap-name":.*`)
```
Comment (Meulengracht) [inline interfaces/mount/backend_test.go:348]: what are expections in terms of the fstab file contents in this case? Do we expect the same fstab file as in the happy case?

**Review by Meulengracht — APPROVED** [review]
Thank you for adding this! I have a question related to verifying the error case

File: interfaces/mount/backend_test.go (hunk 1 above)
Comment (pedronis) [reply interfaces/mount/backend_test.go:348]: I fear so, as there's no undo logic in this context

Comment (Meulengracht) [reply interfaces/mount/backend_test.go:348]: the fstab checks you've added are nice, then it's clear that it's expected

**Review by Meulengracht — APPROVED** [review]
LGTM!

**Review by MiguelPires — APPROVED** [review]
LGTM, thanks

File: interfaces/ifacetest/backendtest.go [hunk 2]
```diff
@@ -211,15 +211,21 @@ func (s *BackendSuite) InstallSnap(c *C, opts interfaces.ConfinementOptions, ins
 
 // UpdateSnap "updates" an existing snap from YAML.
 func (s *BackendSuite) UpdateSnap(c *C, oldSnapInfo *snap.Info, opts interfaces.ConfinementOptions, snapYaml string, revision int) *snap.Info {
```
[the remark this answers is not in the document]
Comment (pedronis) [reply interfaces/ifacetest/backendtest.go:213]: I didn't want to update all preexisting tests

File: interfaces/mount/backend_test.go [hunk 3]
```diff
@@ -246,3 +246,118 @@ func (s *backendSuite) TestSandboxFeatures(c *C) {
...
+	// ensure both security effects from iface/iface2 are combined
+	// (because mount profiles are global in the whole snap)
+	expected = strings.Split(fmt.Sprintf("%s\n%s\n%s\n", fsEntry1, fsEntry2, fsEntry3), "\n")
+	sort.Strings(expected)
+	// and that we have the modern fstab file (global for snap)
+	content, err = ioutil.ReadFile(fn)
+	c.Assert(err, IsNil, Commentf("Expected mount profile for the whole snap"))
+	got = strings.Split(string(content), "\n")
+	sort.Strings(got)
+	c.Check(got, DeepEquals, expected)
```
[the remark this answers is not in the document]
Comment (pedronis) [reply interfaces/mount/backend_test.go:310]: ah, I can look into updating this in a follow up

**Pull request merged**